src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -1021,3 +978,92 @@ EnsureReplicationSettings(Oid relationId, char replicationModel)
 								"factor\" to one%s.", extraHint)));
 	}
 }
+
+
+/*
+ * CopyLocalData copies local data into the shards.
+ */
+static void
+CopyLocalData(Oid relationId)
+{
+	DestReceiver *copyDest = NULL;
+	List *columnNameList = NIL;
+	Relation distributedRelation = NULL;
+	TupleDesc tupleDescriptor = NULL;
+	int columnIndex = 0;
+	bool stopOnFailure = true;
+
+	EState *estate = NULL;
+	HeapScanDesc scan = NULL;
+	HeapTuple tuple = NULL;
+	ExprContext *econtext = NULL;
+	MemoryContext oldContext = NULL;
+	TupleTableSlot *slot = NULL;
+	uint64 rowsCopied = 0;
+
+	distributedRelation = heap_open(relationId, ExclusiveLock);
+	tupleDescriptor = RelationGetDescr(distributedRelation);
+	slot = MakeSingleTupleTableSlot(tupleDescriptor);
+
+	for (columnIndex = 0; columnIndex < tupleDescriptor->natts; columnIndex++)
+	{
+		Form_pg_attribute currentColumn = tupleDescriptor->attrs[columnIndex];
+		char *columnName = NameStr(currentColumn->attname);
+
+		if (currentColumn->attisdropped)
+		{
+			continue;
+		}
+
+		columnNameList = lappend(columnNameList, columnName);
+	}
+
+	estate = CreateExecutorState();
+	econtext = GetPerTupleExprContext(estate);
+	econtext->ecxt_scantuple = slot;
+
+	copyDest =
+		(DestReceiver *) CreateCitusCopyDestReceiver(relationId, columnNameList,
+													 estate, stopOnFailure);
+
+	copyDest->rStartup(copyDest, 0, tupleDescriptor);
+
+	scan = heap_beginscan(distributedRelation, GetActiveSnapshot(), 0, NULL);
```

Comment:
This'll "do the right thing" if there have already been e.g. `UPDATE`s against the local data  in this transaction, right?